root/components/Header.js

```diff
@@ -0,0 +1,46 @@
+/*
+ * @flow
+ * Copyright (C) 2017 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+const React = require('react');
+const {l} = require('../static/scripts/common/i18n');
+const EntityTabs = require('./EntityTabs');
+const SubHeader = require('./SubHeader');
+const EntityLink = require('../static/scripts/common/components/EntityLink');
+const TaggerIcon = require('../static/scripts/common/components/TaggerIcon');
+
+type Props = {
+  entity: EntityT;
+  editTab?: string | React.Node;
+  headerClass: string;
+  heading?: string | React.Node;
+  page: string;
+  subHeading: string;
+};
+
+const Header = ({entity, headerClass, subHeading, ...props}: Props) => {
+  let heading = <EntityLink entity={entity} />;
+
+  if (props.heading) {
+    heading = props.heading;
+  }
+
+  const header = (
+    <div className={headerClass}>
+      {$c.session.tport ? <TaggerIcon entity={entity} /> : null}
+      <h1>{heading}</h1>
+      <SubHeader subHeading={subHeading} />
+    </div>
+  );
+
+  const tabs = <EntityTabs entity={entity} {...props} />;
+
+  return [header, tabs];
+};
```

Comment:
This whole component can be more simply written:

```JavaScript
const Header = ({entity, headerClass, heading, subHeading, ...props}: Props) => (
  <React.Fragment>
    <div className={headerClass}>
      {$c.session.tport ? <TaggerIcon entity={entity} /> : null}
      <h1>
        {heading || <EntityLink entity={entity} />}
      </h1>
      <SubHeader subHeading={subHeading} />
    </div>
    <EntityTabs entity={entity} {...props} />
  </React.Fragment>
);
```

Right now you're allocating an `<EntityLink />` even if it's unused.

`React.Fragment` is similar to returning an array, but doesn't require keys and allows for a consistent syntax.